syntax = "proto3";

package NAlice.NScenarios;

option go_package = "a.yandex-team.ru/alice/megamind/protos/scenarios";
option java_package = "ru.yandex.alice.megamind.protos.scenarios";
option java_outer_classname = "Frame";

import "alice/library/censor/protos/extension.proto";

import "alice/megamind/protos/common/atm.proto";
import "alice/megamind/protos/common/frame.proto";
import "alice/megamind/protos/common/frame_request_params.proto";
import "alice/megamind/protos/common/origin.proto";
import "alice/megamind/protos/common/request_params.proto";

import "mapreduce/yt/interface/protos/extension.proto";

// A frame with an utterance it is assumed to correspond to
message TParsedUtterance {
    option (NYT.default_field_flags) = SERIALIZATION_YT;

    string Utterance = 1 [json_name = "utterance", (NYT.column_name) = "utterance", (FieldAccess) = A_PRIVATE_REQUEST];
    TSemanticFrame Frame = 2 [json_name = "frame", (NYT.column_name) = "frame", deprecated = true];
    TTypedSemanticFrame TypedSemanticFrame = 3 [json_name = "typed_semantic_frame", (NYT.column_name) = "typed_semantic_frame"];
    TAnalyticsTrackingModule Analytics = 4 [json_name = "analytics", (NYT.column_name) = "analytics"];
    TOrigin Origin = 5 [json_name = "origin", (NYT.column_name) = "origin"];
    // TFrameRequestParams is deprecated. Use TRequestParams instead.
    TFrameRequestParams Params = 6 [json_name = "params", (NYT.column_name) = "params", deprecated = true];
    TRequestParams RequestParams = 7 [json_name = "request_params", (NYT.column_name) = "request_params"];
}
